packages/robusta_firebase_cloud_messaging/lib/src/extension.dart

```diff
@@ -0,0 +1,80 @@
+import 'dart:async';
+import 'dart:developer';
+
+import 'package:firebase_messaging/firebase_messaging.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:meta/meta.dart';
+import 'package:robusta_firebase_core/robusta_firebase_core.dart';
+import 'package:robusta_runner/robusta_runner.dart';
+
+part './permission_settings.dart';
+
+/// {@template robusta_firebase_cloud_messaging}
+/// A Helper class for Firebase Cloud Messaiging
+/// {@endtemplate}
+class FirebaseCloudMessagingExtension implements DependenceExtension {
+  /// {@macro robusta_firebase_cloud_messaging}
+  FirebaseCloudMessagingExtension({
+    required EventManager eventManager,
+    NotiPermissionRequestSettings? settings,
+    NotiRequestStrategy? notiRequestStrategy,
```

Comment:
In this context permission request only notification, so prefix with `Noti` is redundance, WDYT?

Reply:
Done.